File: _posts/arrays/best_time_to_buy_and_sell_stock_2/ts/solution_trends.ts
export const maxProfitTrends = (prices: number[]): number => {
    let res = 0

    for (let i = 0; i < prices.length - 1; i++) {
        if (prices[i] < prices[i + 1]) {
            res += prices[i + 1] - prices[i]
        }
    }

    return res
}

File: _posts/arrays/best_time_to_buy_and_sell_stock_2/go/solution_trends.go
package main

func maxProfitTrends(prices []int) int {
	res := 0

	for i := 0; i < len(prices)-1; i++ {
		if prices[i] < prices[i+1] {
			res += prices[i+1] - prices[i]
		}
	}

	return res
}
